mail_notification_custom_subject/models/mail_thread.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2020 Tecnativa - João Marques
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import models, api
+
+
+class MailThread(models.AbstractModel):
+    _inherit = "mail.thread"
+
+    @api.multi
+    @api.returns("mail.message", lambda value: value.id)
+    def message_post(
+        self,
+        body="",
+        subject=None,
+        message_type="notification",
+        subtype=None,
+        parent_id=False,
+        attachments=None,
+        notif_layout=False,
+        add_sign=True,
+        model_description=False,
+        mail_auto_delete=True,
+        **kwargs
+    ):
+        new_message = super().message_post(
```

Comment:
If you call super before, the mail sending is done in this stage AFAIK

Reply:
From my testing the email is only sent after I return the new message
at the end.